modules/standard/Set.chpl

```diff
@@ -627,7 +634,31 @@ module Set {
     :arg rhs: A set to take the intersection of.
   */
   proc &=(ref lhs: set(?t, ?), const ref rhs: set(t, ?)) {
-    lhs = lhs & rhs;
+    /* Iterate over the smaller set.  But we can't remove things from
+       lhs while iterating over it.  So use a temporary if lhs is
+       significantly smaller than rhs; otherwise just iterate over rhs. */
+    if lhs.size < 2 * rhs.size {
+      var result: set(t, (lhs.parSafe || rhs.parSafe));
+
+      if lhs.parSafe && rhs.parSafe {
+        forall x in lhs do
+          if rhs.contains(x) then
+            result.add(x);
+      } else {
+        for x in lhs do
+          if rhs.contains(x) then
+            result.add(x);
+      }
+      lhs = result;
+    } else {
+      if lhs.parSafe && rhs.parSafe {
+        forall x in rhs do
+          lhs.remove(x);
+      } else {
+        for x in rhs do
+          lhs.remove(x);
```

Comment:
I mentioned this to @dlongnecke-cray already, but I think this branch is wrong -- it looks like its just implementing `-=` instead of `&=`

Reply:
Oops, thanks.  I opened  #15958 to cover that.